pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -195,12 +198,20 @@ func (cs *ConntrackConnectionStore) AddOrUpdateConn(conn *flowexporter.Connectio
 	connKey := flowexporter.NewConnectionKey(conn)
 	cs.mutex.Lock()
 	defer cs.mutex.Unlock()
-	existingConn, exists := cs.connections[connKey]
+
+	existingConn, exists := cs.Connections[connKey]
+	currTime := time.Now()
 
 	if exists {
 		existingConn.IsPresent = true
-		// avoid updating stats of the existing connection that is about to close
+		cs.expirePriorityQueue.Mutex.Lock()
 		if flowexporter.IsConnectionDying(existingConn) {
+			// If the connection is in dying state and the corresponding pqItem is already
+			// exported, then update the DyingAndDoneExport flag on the connection.
+			if existingConn.PriorityQueueItem.DyingAndDoneExport {
+				conn.DyingAndDoneExport = true
+			}
+			cs.expirePriorityQueue.Mutex.Unlock()
 			return
 		}
 		// Update the necessary fields that are used in generating flow records.
```

Comment:
Move L217 to L224 above? This way you could keep priority queue related code in one place, so that releasing lock to that becomes simple.

Reply:
I feel there's no need to update existingConn if we find it is `DyingAndDoneExport`. Maybe we can just use lock and defer unlock at the beginning as previous. That will not minimize the critical section but make the lock releasing looks simpler and cleaner. I have no strong opinion on that. What do you think?

Reply:
This is fine.